sql/src/main/java/org/opensearch/sql/sql/antlr/AnonymizerListener.java

```diff
@@ -0,0 +1,107 @@
+package org.opensearch.sql.sql.antlr;
+
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.BACKTICK_QUOTE_ID;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.BOOLEAN;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.COMMA;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.DECIMAL_LITERAL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.DOT;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.EQUAL_SYMBOL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.EXCLAMATION_SYMBOL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.FALSE;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.FROM;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.GREATER_SYMBOL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.ID;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.LESS_SYMBOL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.ONE_DECIMAL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.REAL_LITERAL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.STRING_LITERAL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.TIMESTAMP;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.TRUE;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.TWO_DECIMAL;
+import static org.opensearch.sql.sql.antlr.parser.OpenSearchSQLLexer.ZERO_DECIMAL;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.tree.ErrorNode;
+import org.antlr.v4.runtime.tree.ParseTreeListener;
+import org.antlr.v4.runtime.tree.TerminalNode;
+
+/**
+ * Parse tree listener for anonymizing SQL requests.
+ */
+public class AnonymizerListener implements ParseTreeListener {
+  private String anonymizedQueryString = "";
```

Comment:
I think this string needs to be reset after every query. I'm getting concatenated anonymized queries. For example:
First query: `SELECT time0 FROM calcs`
Anonymized query: `New Engine Request Query: ( SELECT identifier FROM table )`
Second query: `SELECT CAST(time0 AS STRING) FROM calcs`
Anonymized query: `New Engine Request Query: ( SELECT identifier FROM table  SELECT CAST ( identifier AS STRING ) FROM table )`

This could make logs extremely large and could slow down the instance as well.

Reply:
Oops this bug snuck it's way back into my code, fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/266/commits/d6d6dc84afd465cabd9389caaac2b6df205a69df

Reply:
I tested again and bug still persists. Might be good to have a unit test that uses the same AnonymizerListener object. Current tests creates a new one for every test.

Reply:
I have reworked my code a bit in https://github.com/Bit-Quill/opensearch-project-sql/pull/266/commits/559dbbfbaa6613ba67c7c90337c64ac6fda75f79 so it should be working fine now